### PR TITLE
fix(reflection_capability): make injection work on inherited directives

### PR DESCRIPTION
**Pull request by lacolaco**
superseded #8991

### CODE REVIEW

**Comment by laco0416** [comment]
I want to squash commits in this PR, but if I do it original author will be disappeared. Can I do it or any good solutions?

File: modules/playground/src/inherit_from_angular/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class MyService {
+}
+
+@Directive({selector: '[ngIfService]'})
+class NgIfService extends NgIf {
+  constructor(
+      _viewContainerRef: ViewContainerRef, _templateRef: TemplateRef<Object>,
+      myService: MyService) {
+    super(_viewContainerRef, _templateRef);
+    console.log(myService);
```
Comment (vicb) [inline modules/playground/src/inherit_from_angular/index.ts]: debug

**Comment by vicb** [comment]
Keep the original author first commit and squash your commits.

I think we should have more test, including low level ones:
- Make sure a `Component`, `Directive` or `Pipe` decoration on a derived class override the decorator on the base class,
- Make sure a `Pipe`, ... decorator on a derived class does not modify the original metadata,
- Make sur `Input` and `Output` on from the base class does not affect the derived class,
- There should be an integration that runs in both JiT and AoT mode - see `angular/modules/@angular/compiler-cli/integrationtest`

Thanks

**Comment by hannahhoward** [comment]
Hi everyone! Many apologies for my absence -- yes life has moved on and I donut have time to complete this. Can answer questions for whomever is taking on this work.

**Comment by vicb** [comment]
@hannahhoward mo pb, thanks for the initial work on this.

**Comment by laco0416** [comment]
@vicb Bosch seems started adding inheritance support into the compiler layer. so I think this PR is no longer needed. should I close here and wait for that?

 https://github.com/angular/angular/issues/11606#issuecomment-262315256

**Comment by mrgoos** [comment]
Hey guys, any news on that?
Thanks.

**Comment by mlc-mlapis** [comment]
@tbosch If I understand correctly "template/templateUrl" or "styles/styleUrls" on a base component are now ignored from a derived component point of view now.

I read all discussion about the problem (like https://docs.google.com/document/d/16D3U3wdDBMEncUUEXBxiCaI11SnhjBijggZGq8JMjP0) again and I still do not understand why there are difficulties on decission how to inherit "template/templateUrl" or "styles/styleUrls" between a base and a derived component.

What is wrong on the following rules?

- If a derived component does not declare template or styles then both are taken from its base component, if exists,
- if a derived component declares them (in any variant) then definitions are taken from this derived component on the same rules as it would be a normal (base) component.

That is all. A programmer who writes the code has the same chance to make a mistake and it does not matter whether it is on a base or on a derived component.

**Pull request closed without merging**